holoviews/plotting/bokeh/element.py

```diff
@@ -2218,12 +2218,16 @@ def _process_legend(self, plot=None):
             or not self.show_legend):
             legend.items[:] = []
         else:
-            plot.legend.orientation = 'horizontal' if self.legend_cols else 'vertical'
+            if bokeh3 and self.legend_cols:
+                plot.legend.nrows = self.legend_cols
```

Comment:
`nrows` are used here to get the desired effect:

|   |   | 
| -- | -- |
| 0 | 2 |
|1 | 3 | 

If `ncols` are used:
|   |   | 
| -- | -- |
| 0 | 1 |
|2 | 3 |

Reply:
Seems odd, would expect exactly the opposite 🤷